File: src/my_cpp_pkg/src/robot_news_station.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

class RobotNewsStationNode : public rclcpp::Node
{

public:
    RobotNewsStationNode() : Node("robot_news_station")
    {
        publisher=this->create_publisher<example_interfaces::msg::String>("robot_news", 10);
        timer = this->create_wall_timer(std::chrono::seconds(1), std::bind(&RobotNewsStationNode::publish_news, this));
        RCLCPP_INFO(this->get_logger(), "heyo rbot news node");
    }

private:
    rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr publisher;

    void publish_news()
    {
        auto msg = example_interfaces::msg::String();
        msg.data = "herro";
        publisher -> publish(msg);
    }
    rclcpp::TimerBase::SharedPtr timer;
};

int main(int argc, char **argv)
{

    rclcpp::init(argc, argv);
    auto node = std::make_shared<RobotNewsStationNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src/my_py_pkg/my_py_pkg/robot_news_station.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node
from example_interfaces.msg import String


class MyNode(Node):
    def __init__(self):
        super().__init__("robot_news_station")
        self.publisher = self.create_publisher(String, "robot_news", 10)
        self.timer = self.create_timer(0.5, self.publish_news)
        self.get_logger().info("ROBOT NEWS INITIATED")

    def publish_news(self):
        msg = String()
        msg.data = "HELLLO"
        self.publisher.publish(msg)


def main(args=None):
    rclpy.init(args=args)
    node = MyNode()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()
